### PR TITLE
Accepting buoyancy instead of density in `MultiLayerQG` module

### PR DESCRIPTION
**Pull request by mpudig**
This pull request addresses issue #339 and discussion #325.

The conclusion from the discussion was that we should accept buoyancy as an input instead of density in `multilayerqg`. The density array was used only to calculate the reduced gravity array. Previously, the derived reduced gravity array resulted in spurious internal PV gradients as discussed extensively in discussion #325. It also implied that the Boussinesq approximation did not hold, which is implicitly assumed in the form of the layered QG equations solved here.

This pull request has changed the input field of `multilayerqg` from $\rho$, the density, to $b = -g \frac{\delta \rho}{\rho_0}$, the Boussinesq buoyancy, where $|\delta \rho| \ll \rho_0$. The reduced gravity is now derived directly from $b$, https://github.com/mpudig/GeophysicalFlows.jl/blob/density_to_buoyancy/src/multilayerqg.jl#L348. I have set the change in buoyancy to scale like 1/nlayers with the other default parameters as they are in `multilayerqg` – I believe this is consistent with the other nondimensional parameter values and an order 1 Burger number.

### CODE REVIEW

**Comment by navidcy** [comment]
Thanks for this! I'll try to get to this sometime this week!

File: src/multilayerqg.jl [hunk 1]
```diff
@@ -92,14 +90,13 @@ function Problem(nlayers::Int,                             # number of fluid lay
...
-                            U = zeros(nlayers),            # imposed zonal flow U(y) in each layer
-                            H = 1/nlayers * ones(nlayers), # rest fluid height of each layer
-                            ρ = Array{Float64}(1:nlayers), # density of each layer
-                          eta = nothing,                   # periodic component of the topographic PV
-      topographic_pv_gradient = (0, 0),                    # tuple with the ``(x, y)`` components of topographic PV large-scale gradient
+                           f₀ = 1.0,                                             # Coriolis parameter
+                            β = 0.0,                                             # y-gradient of Coriolis parameter
+                            U = zeros(nlayers),                                  # imposed zonal flow U(y) in each layer
+                            H = 1/nlayers * ones(nlayers),                       # rest fluid height of each layer
+                            b = -(1 .+ 1/nlayers * Array{Float64}(0:nlayers-1)), # Boussinesq buoyancy of each layer
```
Comment (navidcy) [inline src/multilayerqg.jl:97]: any reasoning behind this default `b`?

Comment (navidcy) [reply src/multilayerqg.jl:97]: is this what corresponds to the previous default ρ?

Comment (navidcy) [reply src/multilayerqg.jl:97]: I think it's the same if ρ0 = ρ[n], right?

**Review by navidcy — APPROVED** [review]
lgtm! nice job!

**Comment by navidcy** [comment]
@mpudig you should have merge rights now, right?

if you are happy merge whenever.

**Comment by navidcy** [comment]
@mpudig I'll merge

**Comment by navidcy** [comment]
@mpudig welcome to GeophysicalFlows.jl contributors!

**Comment by mpudig** [comment]
@navidcy Apologies for the delay, I was making the long trip back to Australia. Thanks for merging!!

File: src/multilayerqg.jl (hunk 1 above)
Comment (mpudig) [reply src/multilayerqg.jl:97]: For the two-layer model this choice of default $b$ will give the same reduced gravity as the previous default $\rho$ did. I chose this as the default $b$ as I believe it's consistent with QG scalings.

**Pull request merged**